serving/src/main/java/ai/djl/serving/plugins/DependencyManager.java

```diff
@@ -0,0 +1,235 @@
+                    installDependency(
+                            depDir, "com.microsoft.onnxruntime:onnxruntime_gpu:" + ortVersion);
+                } else {
+                    installDependency(
+                            depDir, "com.microsoft.onnxruntime:onnxruntime:" + ortVersion);
+                }
+                break;
+            case "PaddlePaddle":
+                installDependency(depDir, "ai.djl.paddlepaddle:paddlepaddle-engine:" + djlVersion);
+                break;
+            case "TFLite":
+                installDependency(depDir, "ai.djl.tflite:tflite-engine:" + djlVersion);
+                break;
+            case "XGBoost":
+                installDependency(depDir, "ai.djl.ml.xgboost:xgboost-engine:" + djlVersion);
+                break;
+            case "DLR":
+                installDependency(depDir, "ai.djl.dlr:dlr-engine:" + djlVersion);
+                break;
+            default:
+                break;
+        }
+
+        // refresh EngineProvider
+        MutableClassLoader mcl = MutableClassLoader.getInstance();
+        for (EngineProvider provider : ServiceLoader.load(EngineProvider.class, mcl)) {
+            Engine.registerEngine(provider);
+        }
+
+        // refresh ZooProvider
+        for (ZooProvider provider : ServiceLoader.load(ZooProvider.class, mcl)) {
+            ModelZoo.registerModelZoo(provider);
+        }
+    }
+
+    private synchronized void installDependency(Path depDir, String dependency) throws IOException {
+        String[] tokens = dependency.split(":");
+        if (tokens.length < 3) {
+            logger.info("Invalid dependency: {}", dependency);
+            return;
+        }
+
+        logger.info("Loading dependency: {}", dependency);
+        String groupId = tokens[0].replace('.', '/');
+        String artifactId = tokens[1];
+        String version = tokens[2];
+        String name;
+        if (tokens.length == 3) {
+            name = artifactId + '-' + version + ".jar";
+        } else {
+            name = artifactId + '-' + version + '-' + tokens[3] + ".jar";
+        }
+
+        Path file = depDir.resolve(name);
+        if (Files.isRegularFile(file)) {
+            logger.info("Found existing dependency: {}", name);
+        } else {
+            String maven = "https://search.maven.org/remotecontent?filepath=";
+            String link = maven + groupId + '/' + artifactId + '/' + version + '/' + name;
+            logger.info("Downloading dependency: {}.", link);
+            Path tmp = depDir.resolve(name + ".tmp");
+            try (InputStream is = new URL(link).openStream()) {
+                Files.copy(is, tmp);
+                Utils.moveQuietly(tmp, file);
+            } finally {
+                Files.deleteIfExists(tmp);
+            }
+        }
+        MutableClassLoader mcl = MutableClassLoader.getInstance();
+        mcl.addURL(file.toUri().toURL());
+    }
+
+    private static String resolveDjlVersion() {
```

Comment:
This queries for the latest version, but don't we need the version that is currently being used?

Reply:
The active DJL version is SNAPSHOT version, which we cannot be downloaded from maven central. To download SNAPSHOT version is complicated. The logic here is a workaround for development.